common/scala/src/main/scala/org/apache/openwhisk/core/entity/Parameter.scala

```diff
@@ -157,14 +174,19 @@ protected[entity] class ParameterName protected[entity] (val name: String) exten
  * @param v the value of the parameter, may be null
  * @param init if true, this parameter value is only offered to the action during initialization
  */
-protected[entity] case class ParameterValue protected[entity] (private val v: JsValue, val init: Boolean) {
+protected[entity] case class ParameterValue protected[entity] (private val v: JsValue,
+                                                               val init: Boolean,
+                                                               val e: JsValue = JsNull) {
```

Comment:
Why not just store the encrypted value and a boolean to indicate it's encrypted?

Reply:
Adding which scheme would allow someone later to migrate from scheme A to B if an issue surfaces in the former.

Reply:
should `e` be an option type (which would allow omission) - similarly would it be simpler if when present this value is meaningful (a string or bool and not null).

at this point i'm not sure what values this takes on yet.

Reply:
Yup, I went with your suggestion and locked down what the encryption schemes are allowed to do.